nifi-nar-bundles/nifi-standard-services/nifi-dbcp-service-bundle/nifi-db-schema-registry-service/src/test/java/org/apache/nifi/db/schemaregistry/DatabaseTableSchemaRegistryTest.java

```diff
@@ -0,0 +1,224 @@
+                    " code integer CONSTRAINT CODE_RANGE CHECK (code >= 0 AND code < 1000), dt date)",
+            "CREATE TABLE SCHEMA2.PERSONS (id2 integer primary key, name varchar(100)," +
+                    " code integer CONSTRAINT CODE_RANGE CHECK (code >= 0 AND code < 1000), dt date)",
+            "CREATE TABLE UUID_TEST (id integer primary key, name VARCHAR(100))",
+            "CREATE TABLE LONGVARBINARY_TEST (id integer primary key, name LONG VARCHAR FOR BIT DATA)"
+    );
+
+    private static final String SERVICE_ID = DBCPServiceSimpleImpl.class.getName();
+
+    private final static String DB_LOCATION = "target/db_schema_reg";
+
+    // This is to mimic those in DBCPProperties to avoid adding the dependency to nifi-dbcp-base
+    private static final PropertyDescriptor DATABASE_URL = new PropertyDescriptor.Builder()
+            .name("Database Connection URL")
+            .addValidator(new ConnectionUrlValidator())
+            .expressionLanguageSupported(ExpressionLanguageScope.ENVIRONMENT)
+            .build();
+
+    private static final PropertyDescriptor DB_USER = new PropertyDescriptor.Builder()
+            .name("Database User")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.ENVIRONMENT)
+            .build();
+
+    private static final PropertyDescriptor DB_PASSWORD = new PropertyDescriptor.Builder()
+            .name("Password")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.ENVIRONMENT)
+            .build();
+    private static final PropertyDescriptor DB_DRIVERNAME = new PropertyDescriptor.Builder()
+            .name("Database Driver Class Name")
+            .addValidator(new DriverClassValidator())
+            .expressionLanguageSupported(ExpressionLanguageScope.ENVIRONMENT)
+            .build();
+
+    private TestRunner runner;
+
+    @BeforeAll
+    public static void setupDatabase() throws Exception {
+        System.setProperty("derby.stream.error.file", "target/derby.log");
+
+        // remove previous test database, if any
+        final File dbLocation = new File(DB_LOCATION);
+        try {
+            FileUtils.deleteFile(dbLocation, true);
+        } catch (IOException ignore) {
+            // Do nothing, may not have existed
+        }
+
+        // Mock the DBCP Controller Service so we can control the Results
+        DBCPService setupDbService = spy(new DBCPServiceSimpleImpl(DB_LOCATION));
+
+        final Connection conn = setupDbService.getConnection();
+        final Statement stmt = conn.createStatement();
+        for (String createTableStatement : CREATE_TABLE_STATEMENTS) {
+            stmt.execute(createTableStatement);
+        }
+
+        stmt.close();
+        conn.close();
+    }
+
+    @AfterAll
+    public static void shutdownDatabase() throws Exception {
```

Comment:
The `Exception` is not thrown and can be removed:
```suggestion
    public static void shutdownDatabase() {
```